src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -118,7 +117,7 @@ void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             var namingStyleRules = namingPreferences.Rules;
 
             if (!namingStyleRules.TryGetApplicableRule(symbol, out var applicableRule) ||
-                applicableRule.EnforcementLevel.WithDefaultSeverity(DiagnosticSeverity.Hidden) >= ReportDiagnostic.Hidden)
```

Comment:
what does this mean?

Reply:
If the `ReportDiagnostic` has a value of `Default`, `Hidden`, or `Suppress`, the expression on this line is true.

* `Default` maps to `Hidden` per the call to `WithDefaultSeverity`
* `Hidden` and `Suppress` are the only two values greater than or equal to `Hidden` (this enumeration is defined in the opposite order from `DiagnosticSeverity`)